### ISSUE TITLE
Could not find a valid gem 'rest-client'

### ISSUE THREAD

**Issue by warriorkitty** [topofocus/active-orient#2]
```bash
=>:~$ sudo gem install active-orient
ERROR:  Could not find a valid gem 'rest-client' (~> 2.0) in any repository
ERROR:  Possible alternatives: rest-client
```

**Comment by topofocus** [comment]
Hi,
1. you need a ruby 2.1or 2.2 environment to run ActiveOrient.
Please consider to use rvm.
Then all gems are stored in ~/.rvm/(ruby-version).gems
2. Are you using the 0.4-gem? 
The Gemfile in the most recent version shows
gem 'rest-client', :git => 'git://github.com/rest-client/rest-client.git'
If you edit the Gemfile accordingly, this should work.

3. Its advisable to use the bundler to manage gems.
The Gemfile might look as follows
```
 1 source "https://rubygems.org"
  2 gem 'activesupport'
  3 gem 'activemodel'
  4 gem 'mail'
  5 gem 'xml-simple'
  6 gem 'getopt'
  7 gem 'gnuplot'
  8 gem 'active-orient' , "~>0.3" , :git =>  'git://github.com/topofocus/active-orient.git'
  9 gem 'descriptive_statistics', :git => 'git://github.com/thirtysixthspan/descriptive_statistics.git'
 10 group :development, :test do
 11 #       gem "factory_girl", "~> 4.0"
 12         gem "rspec"
 13         gem 'rspec-its'
 14         gem 'rspec-collection_matchers'
 15         gem 'rspec-context-private'
 16         gem 'guard'
 17         gem 'guard-rspec'
 18 ##      gem 'database_cleaner'
 19         gem 'rb-inotify'
 20         gem 'pry'
 21 end
 22 gem 'ib-ruby', :git=> 'git://github.com/topofocus/ib-ruby.git'
```

Then to start create a config/boot.rb

```
  1 require 'bundler/setup'
  2# require 'ib-ruby'
  3 require 'active-orient'
(...)
 11 project_root = File.expand_path('../..', __FILE__)
 12 
 13 require File.expand_path('../../lib/support', __FILE__)
(and so on)
```

to test the installation, you may open an irb-session in your working-directory and
require './config/boot'

Then anything should work fine.


I am working on a replacement of the rest-client with the native binary driver now.

hartmut

**Comment by warriorkitty** [comment]
Thank you for a fast response! I had some problems with RVM on my Linux distro (and I just needed the newest Ruby always) but I will try it again.

Thank you for your great work.

**Comment by tommasop** [comment]
@topofocus I tried your hack but it is not working. It complains about Syntax errors in rest.rb file:

```
     irb(main):001:0> require './config/boot'
     SyntaxError: /Users/tommasop/code/work/altro/active-orient/lib/rest.rb:359: syntax error, unexpected ','
..._class , attributes: {}, from:,  to:, unique: false
...                               ^
/Users/tommasop/code/work/altro/active-orient/lib/rest.rb:480: syntax error, unexpected ','
    def create_index o_class  , name:, on: :automatic, type: :unique
                                      ^
/Users/tommasop/code/work/altro/active-orient/lib/rest.rb:889: syntax error, unexpected ','
    def update_documents o_class, set: , where: {}
                                        ^
/Users/tommasop/code/work/altro/active-orient/lib/rest.rb:892: syntax error, unexpected keyword_end, expecting end-of-input
	from /Users/tommasop/code/work/altro/active-orient/config/boot.rb:10:in `require'
	from /Users/tommasop/code/work/altro/active-orient/config/boot.rb:10:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from (irb):1
	from /usr/bin/irb:12:in `<main>'
```

**Comment by topofocus** [comment]
First, please use 
 gem 'active-orient' , "~>0.4" , :git =>  'git://github.com/topofocus/active-orient.git'.
in your gemfile.
I added some improvements to the query-lib and realized a substantial performance enhancement in the base-class. 

Second, most probably you are using "ruby 2.0.0". The «required keyword parameter«- Syntax was introduced in Ruby 2.1.

**Comment by tommasop** [comment]
You are perfectly right with ruby 2.2.3 the error goes away, still if I use the 0.4 version like this:

```
source "https://rubygems.org"
  1 gem 'activesupport'
  2 gem 'activemodel'
  3 gem 'mail'
  4 gem 'xml-simple'
  5 gem 'getopt'
  6 gem 'gnuplot'
  7 gem 'active-orient' , "~>0.4" , :git =>  'git://github.com/topofocus/active-orient.git'
  8 gem 'descriptive_statistics', :git => 'git://github.com/thirtysixthspan/descriptive_statistics.git'
  9 group :development, :test do
 10   #       gem "factory_girl", "~> 4.0"
 11   gem "rspec"
 12   gem 'rspec-its'
 13   gem 'rspec-collection_matchers'
 14   gem 'rspec-context-private'
 15   gem 'guard'
 16   gem 'guard-rspec'
 17   ##      gem 'database_cleaner'
 18   gem 'rb-inotify'
 19   gem 'pry'
 20 end
 21 gem 'ib-ruby', :git=> 'git://github.com/topofocus/ib-ruby.git'
```

I have the following error:

```
irb(main):001:0> require './config/boot'
Could not find active-orient-0.4 in any of the sources
Run `bundle install` to install missing gems.
```

And if I try gem install active-orient I receive:

```
gem install active-orient
ERROR:  Could not find a valid gem 'rest-client' (~> 2.0) in any repository
ERROR:  Possible alternatives: rest-client
```

Do I need to build and install the gem locally?

Thanks for your support

**Comment by topofocus** [comment]
Hi,
I just published a bugfix release gem (0.41).
Please consider using this. it refers to the version 1.8 of the rest-client.

**Comment by tommasop** [comment]
Thanks for your great support! It's working
